### PR TITLE
Misc fixes around resource display

### PR DESCRIPTION
**Pull request by clumens**
This is the easy stuff from #2413 pulled out into its own PR.

### CODE REVIEW

File: lib/common/strings.c [hunk 1]
```diff
@@ -872,14 +872,30 @@ pcmk__parse_ll_range(const char *srcstring, long long *start, long long *end)
...
- * \param[in]  s    String to search for
+ * Behavior can be changed with various flags:
+ *
+ * - pcmk__str_casei - By default, comparisons are done taking case into
+ *                     account.  This flag makes comparisons case-insensitive.
+ * - pcmk__str_null_matches - If the input string is NULL, return TRUE.
+ *
+ * \note The special "*" matching rule takes precedence over flags.  In
+ *       particular, "*" will match a NULL input string even without
+ *       pcmk__str_null_matches being specified.
```
Comment (kgaillot) [inline lib/common/strings.c:883]: Not a suggestion for anytime soon, but I wonder if pcmk__str_star_matches would make sense :)

File: lib/pengine/clone.c [hunk 2]
```diff
@@ -933,8 +933,8 @@ pe__clone_text(pcmk__output_t *out, va_list args)
         return rc;
     }
 
-    print_everything = pcmk__str_in_list(only_rsc, rsc_printable_id(rsc)) ||
-                       (strstr(rsc->id, ":") != NULL && pcmk__str_in_list(only_rsc, rsc->id));
+    print_everything = pcmk__str_in_list(only_rsc, rsc_printable_id(rsc), false) ||
+                       (strstr(rsc->id, ":") != NULL && pcmk__str_in_list(only_rsc, rsc->id, false));
```
Comment (kgaillot) [inline lib/pengine/clone.c]: Here and a few below are more places that use true/false

File: lib/fencing/st_output.c [hunk 3]
```diff
@@ -47,7 +47,7 @@ stonith__failed_history(pcmk__output_t *out, va_list args) {
             continue;
         }
 
-        if (!pcmk__str_in_list(only_node, hp->target)) {
+        if (!pcmk__str_in_list(only_node, hp->target, pcmk__str_casei)) {
```
Comment (kgaillot) [inline lib/fencing/st_output.c:50]: I would use only pcmk__str_none in this commit, then do a separate commit setting pcmk__str_casei as a bug fix commit

File: lib/common/tests/strings/pcmk__str_in_list_test.c [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+static void
+in_list(void) {
+    GList *list = NULL;
+
+    list = g_list_prepend(list, (gpointer) "xxx");
+    list = g_list_prepend(list, (gpointer) "yyy");
+
+    g_assert_true(pcmk__str_in_list(list, "xxx", pcmk__str_none));
+    g_assert_true(pcmk__str_in_list(list, "XXX", pcmk__str_casei));
```
Comment (kgaillot) [inline lib/common/tests/strings/pcmk__str_in_list_test.c:77]: Create a list of 3 strings, and make sure we can match at the beginning, middle, and end

File: lib/common/tests/strings/pcmk__str_in_list_test.c [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+static void
+not_in_list(void) {
+    GList *list = NULL;
+
+    list = g_list_prepend(list, (gpointer) "xxx");
+    list = g_list_prepend(list, (gpointer) "yyy");
+
+    g_assert_false(pcmk__str_in_list(list, "XXX", pcmk__str_none));
+    g_assert_false(pcmk__str_in_list(list, "zzz", pcmk__str_none));
+    g_assert_false(pcmk__str_in_list(list, "zzz", pcmk__str_casei));
```
Comment (kgaillot) [inline lib/common/tests/strings/pcmk__str_in_list_test.c:91]: also worth testing that we don't have false positives for substring matches

**Review by kgaillot — COMMENTED** [review]
Looks good, minor comments

File: lib/common/tests/strings/pcmk__str_in_list_test.c (hunk 5 above)
Comment (clumens) [reply lib/common/tests/strings/pcmk__str_in_list_test.c:91]: Good idea, added.

File: lib/fencing/st_output.c (hunk 3 above)
Comment (clumens) [reply lib/fencing/st_output.c:50]: Fixed.

File: lib/pengine/clone.c (hunk 2 above)
Comment (clumens) [reply lib/pengine/clone.c]: Fixed.
